### PR TITLE
Update setup-envtest dependency

### PR DESCRIPTION
**Pull request by chrisdoherty4**
Update `setup-envtest` to patch a vulnerability in an implicit dependency.

### CODE REVIEW

File: hack/tools/go.mod [hunk 1]
```diff
@@ -2,4 +2,7 @@ module github.com/aws/eks-anywhere/hack/tools
 
 go 1.16
```
[the remark this answers is not in the document]
Comment (TerryHowe) [reply hack/tools/go.mod:3]: How big is the setup-envtest dependency. Seems heavy handed, especially if you e2e

[the remark this answers is not in the document]
Comment (chrisdoherty4) [reply hack/tools/go.mod:3]: We should update the minimum version also.

`setup-envtest` is really an integration testing tool that launches a real API server. Its great to get faster feedback than E2E and the testing can be more thorough comparatively with E2E so I think its a great confidence building block.

I don't think we should be running it with unit tests and agree its heavy handed in that context. I think there are numerous unit tests we have that shouldn't be unit tests though :shrug:.

**Review by TerryHowe — APPROVED** [review]
/lgtm

**Pull request merged**